### ISSUE TITLE
Persist Contentlets as Json

### ISSUE THREAD

**Issue by nollymar** [dotCMS/core#21011]
As a user I would like to be able to store a JSON representation of a piece of content in a database. This would provide the possibility of querying the database against the json object and having a more readable representation of the content. This is a first approach of how the json would look like: https://gist.github.com/wezell/142a905195de6e8e3aae34252cf2b362

Some important aspects to be considered for the implementation:
- A new column should be created in the `contentlet` table to store the json. This column will be optional. The datatype for each database would be:

  - Postgres: JSONB
  - Oracle: NCLOB
  - MySQL: text
  - MSSQL: NVARCHAR(MAX) 

#### CHANGES ON THIS CARD ARE ONLY FOR POSTGRES

- There must be a logic to create an immutable Contentlet object from the json column and viceversa, a logic to generate an immutable json given a Contentlet. 
- As the db column is optional, in case it is null or empty, we should keep using our current logic to create contentlets, namely, using the legacy db columns.
- So far, no upgrade tasks will be implemented to populate the json column in database.  
- To prevent potential collisions, all internal properties a piece of content has (i.e.: modDate, modUser, inode, identifier, languageId, showOnMenu, etc) should be stored inside a `dotSystemFields` node. For example: 
```
"dotSystemFields":{
        "inode":"inodeId",
        "identifier":"identifierId",
        "contentType":"contentTypeId",
        "modDate":"2021-10-01 13:17 -0.0",  
        "baseType":"fileAsset",
        "showOnMenu":false,
        "modUser":"dotcms.org.1",
        "sortOrder":0,
        "languageId":1,
        "editState":"disabledWysiwyg",
        "friendlyName":"this is a friendly name",
         //system level tags (all content can have tags)
        "tags":["tag1", "tag2"]
    },
```
- Related content will be stored as an array of identifiers

**Comment by fabrizzio-dotCMS** [comment]
Note for QA: 
So we're basically changing how we store the contentlet and here are some of the things that we need to validate. 
This change affects pretty much all content types. Currently, if you start with a fresh install all the contentlets will still be stored using the old static columns. But any new instance saved will store the values of their fields in a new column  called in the contenlet table called `contentlet_as_json` 

This change comes to remove the hard limit we had before that forced users to have at most 25 fields of the same type.
here are some things we need to test: 

- We need to test that we can have as many fields as we want of any type
- We need to verify we can create any combinations of fields within a Content-type and save and recover the content as expected. 
- We need to test different combinations of the data type supported by the fields e.g. We need to be able to use Numeric Text fields, Decimal Text Fields, and Regular Text Fields.  The same for other fields with a specific data type like Radios etc.. 
- We need to verify our endpoints return content as they should
- We need to test that fields of type `KeyValue` preserve the order of their elements
- We need to test that we can modify Content-types and see the changes reflect on all the instances for the content-type being modified 
- We need to check the generated JSON and make sure the fields we added are there. But do not expect system fields (system fields are not stored in the static columns) They are stored outside the contentlet. Like in a separate table for example categories or in the file system like binaries. 
- Test when a host gets renamed the template field in the JSON gets renamed as well.

**Comment by nollymar** [comment]
**Internal QA:** In general, the functionality works correctly. I found a few issues registered on this [card](https://github.com/dotCMS/core/issues/21356)

**Comment by bryanboza** [comment]
Fixed, tested on release-21.12. I played a little bit with content types and a lot of combinations of files and every field type is working as expected, some new cards reported  related to this change
